apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx

```diff
@@ -42,7 +43,7 @@ export default function SurveyEditor({
   const [activeQuestionId, setActiveQuestionId] = useState<string | null>(null);
   const [localSurvey, setLocalSurvey] = useState<TSurvey | null>();
   const [invalidQuestions, setInvalidQuestions] = useState<String[] | null>(null);
-
+  const [product, setproduct] = useState<TProduct>(productInfo);
```

Comment:
I think we should keep it like we did with the survey:
Pass `product` as a prop and rename the product in the state to `localProduct`. It still might not be ideal but at least this way it's consistent.